packages/core-transaction-pool/lib/guard.js

```diff
@@ -78,22 +78,6 @@ module.exports = class TransactionGuard {
    * @return {Array}
    */
   __cacheTransactions(transactions) {
-    // Edge case: B -> C, A -> B, B -> C
-    // B -> C is added to cache initially, but invalid because no funds.
-    // A -> B is valid, B gets enough funds
-    // B -> C doesn't enter pool again, because it is in cache.
-    // So we don't want to cache a tx when it cannot be applied.
-    // NOTE: will be refactored in 2.1
-    transactions = transactions.filter(transaction => {
-      const errors = []
-      if (!this.pool.walletManager.canApply(transaction, errors)) {
-        this.__pushError(transaction, 'ERR_APPLY', JSON.stringify(errors))
-        return false
-      }
-
-      return true
-    })
```

Comment:
Removal of this code would reintroduce the bug it was fixing, described in the comment?

Reply:
The whole PR is around this bug, as B->C cant enter the pool anymore until A->B forged

Reply:
I could have missed something but I think after this patch we have the following problem:

* New transaction tx1 A->B is received and at that time A does not have enough funds, so it is not allowed to enter the pool, but is however added to the cache.
* At a later time A receives funds by some other transactions, which are also forged.
* We receive tx1 again. This time A has enough funds. However we reject tx1 because it is in the cache.

Reply:
How about when you receive a new block with transactions, extract all senders and receivers and delete them from the cache. Is this even possible or you're caching only the txid?

Reply:
Only ids :/